src/Vault.sol

```diff
@@ -364,22 +384,35 @@ contract Vault is ERC4626, ERC20Permit, ILiquidationSource, Ownable {
 
     _prizePool.contributePrizeTokens(address(this), _amountIn);
 
-    if (_yieldFeeRecipient != address(0) && _yieldFeePercentage != 0) {
-      // We get the percentage of liquidated yield and mint the equivalent amount in fees.
-      _mint(
-        _yieldFeeRecipient,
-        _getYieldFeeAmount(
-          _getAvailableYieldBalance(),
-          (_amountOut * _yieldFeePercentage) / _liquidableYield
-        )
-      );
+    if (_yieldFeePercentage != 0) {
+      uint256 _yieldFeeShares = (_amountOut * FEE_PRECISION) /
+        (FEE_PRECISION - _yieldFeePercentage) -
+        _amountOut;
+
+      _yieldFeeBalances[_yieldFeeRecipient] += _yieldFeeShares;
+      _yieldFeeTotalSupply += _yieldFeeShares;
```

Comment:
Let's just track a single balance eh? I don't think they'll be changing recipients so often that we need a mapping.

Having to track per-recipient fees adds more than 7k to the gas cost of liquidation

Reply:
Done: https://github.com/pooltogether/v5-vault/pull/6/commits/65ec5dd4559e77c24978642cbdc874f1451e5295